app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,24 @@
+class LikesController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    @like = current_user.likes.new(post_id: params[:post_id])
+    if @like.save
+      @like.increment_post_likes
+      redirect_back fallback_location: root_path, notice: 'Like added!'
+    else
+      redirect_back fallback_location: root_path, alert: 'Failed to add like!'
+    end
+  end
```

Comment:
This is okay, but you can try to do this using turbo streams.

Instead of redirect back, do a respond_to with format.turbo_stream. There are two ways you can respond with turbo stream
#
#### First way
```ruby
respond_to do |format|
    format.turbo_stream
    format.html
end
```

By this way, you don't have to pass any parameter to the turbo_stream format, but you will need a file named create.turbo_stream.erb, where you tell turbo_stream which turbo frame update.

#
#### Second way
```ruby
respond_to do |format|
    format.turbo_stream do
      render turbo_stream: turbo_stream.replace(:likes, partial: 'shared/likes', locals: { like: @like } )
    end
    format.html
end
```

Using this way, you don't have to create any file since you are replacing the turbo_frame wit id 'likes' with 'shared/likes' partial, and passing like local variable.

You can find more info in:
- https://turbo.hotwired.dev/handbook/streams#streaming-from-http-responses
- https://rubydoc.info/github/hotwired/turbo-rails/Turbo%2FStreamsHelper:turbo_stream

Reply:
Thx for the info!

Reply:
If you opt to use turbo_stream, you can do it in another PR